Data_cleaning/load_clean_concat.ipynb

```diff
@@ -200,15 +295,36 @@
     "    for file_name in file_names:\n",
     "        df = pd.read_csv(data_folder+file_name+'.csv', parse_dates=['TIMESTAMP'])\n",
     "        df.rename(mapper=remove_pit_suffix, axis='columns', inplace=True)\n",
-    "            \n",
+    "        \n",
     "        if print_text_result: print(f'===== {file_name} =====')\n",
+    "        \n",
+    "        diff_dict = dict()\n",
+    "        prune_unnecessary_features(df)\n",
+    "        \n",
+    "        diff_dict['dates_removed'] = filter_dates(df, file_name)\n",
+    "        diff_dict['duplicates_removed'] = filter_duplicate_date(df)\n",
+    "        \n",
+    "        # here is where the wavelet coefficients are calculated \n",
+    "        df = fill_na_redox_values(df, file_name)\n",
+    "        df = calculate_wavelet_coefficients(df, 1/2, 5, file_name)\n",
+    "        \n",
+    "        window_sizes = get_time_windows([12, 24])\n",
+    "        \n",
+    "        td = timedelta(days = 0, hours = 0, minutes = 5, seconds = 0, milliseconds= 0, weeks = 0)\n",
+    "        \n",
+    "        if 'VII' not in file_name:\n",
+    "            df = sigma_feature_engineering(df, window_sizes=window_sizes, td=td)\n",
+    "        \n",
+    "        diff_dict['missing_data_removed'] = filter_missing_values(df, file_name)\n",
+    "\n",
+    "        # instead of bundling the filtering functions they are applied one by one as the wavelet calculations wedged in \n",
+    "        # stats = filter_df(df, file_name)\n",
     "\n",
-    "        stats = filter_df(df, file_name)\n",
     "        # TODO add log scale to redox and error flag for each redox depth\n",
     "        df = fix_types(df, file_name)\n",
```

Comment:
This could be run after function call prune_unnecessary_features, so as very first changes to the data. Just to make sure that when adding new features the data is handled in correct format.

Reply:
It works as it is, actually I tried to move it higher but it generated an error so I thought it more practical to leave it as it is